docs/source/Howtos/Extending-the-REST-API.md

```diff
@@ -0,0 +1,287 @@
+
+Just remember: `worn = serializers.SerializerMethodField()` is how the API knows to use `get_worn`, and `Meta.fields` is the list of fields that will actually make it into the final JSON.
+
+## Using your serializer
+
+Now let's go back to our `views.py` - at this point, it should look something like this:
+
+```python
+"""
+<mygame>/web/api/views.py
+
+Customized views for the REST API
+"""
+# we'll need these from django's rest framework to make our view work
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework import status
+
+# this implements all the basic Evennia Object endpoint logic, so we're inheriting from it
+from evennia.web.api.views import ObjectDBViewSet
+
+# and we need this to filter our character view
+from evennia.objects.objects import DefaultCharacter
+
+# our own custom view
+class CharacterViewSet(ObjectDBViewSet):
+	"""
+	A customized Character view that adds an inventory detail
+	"""
+	queryset = DefaultCharacter.objects.all_family()
+
+	@action(detail=True, methods=["get"])
+	def inventory(self, request, pk=None):
+		return Response("your inventory", status=status.HTTP_200_OK )
+```
+
+We want to import our new serializer up with the rest of the imports:
+```python
+from .serializers import InventorySerializer
+```
+
+And then, use it in our view:
+```python
+	@action(detail=True, methods=["get"])
+	def inventory(self, request, pk=None):
+		obj = self.get_object()
+		return Response( InventorySerializer(obj).data, status=status.HTTP_200_OK )
+```
+
+That'll use our new serializer to get our character's inventory! Except... not quite. Go ahead and try it: `evennia reboot` and then `/api/characters/1/inventory` like before. Unlike before, you should get an error saying you don't have permission.
+
+## Customizing API permissions
+
+Evennia comes with its own custom API permissions class, connecting the API permissions to the in-game permission hierarchy and locks system. Since we're trying to access the object's data now, we need to pass the `has_object_permission` check as well as the general permission check - and that permission class hardcodes the actions into the object permission checks.
+
+So, since we've added a new action - `inventory` - to our characters endpoint, we need to use our own custom permissions on our characters endpoint. Create one more module file: `<mygame>/web/api/permissions.py`
+
+Like with the previous classes, we'll be inheriting from the original and extending it to take advantage of all the work Evennia already does for us.
+
+```python
+from evennia.web.api.permissions import EvenniaPermission
```

Comment:
Even though it's mentioned on the lines above, I find it's a good idea is to repeat just what file this code should go in. Many just browse the code and don't read the main text unless they have to.

```` 
```python 
# in mygame/web/api/permissions.py

from evennia.web.api.permissions import EvenniaPermission
#...
````